internal/controller/resource_manager.go

```diff
@@ -144,74 +147,124 @@ func getErrFromWatchEvent(event watch.Event) error {
 // watchEvents watches Kubernetes resources. The second part of the of the "list
 // and watch" strategy.
 // Any occurring errors are sent through errChanWatch. The watch stops when ctx is cancelled.
-func (r *ResourceManager) watchEvents(ctx context.Context, apiGroup string, watchOptions metav1.ListOptions, mutationChan chan *pb.KubernetesResourceMutation) error {
-	logger := r.logger.With(zap.String("api_group", apiGroup))
+func (r *ResourceManager) watchEvents(ctx context.Context, resourceVersion string, mutationChan chan *pb.KubernetesResourceMutation) error {
+	logger := r.logger
 
-	objGVR := schema.GroupVersionResource{Group: apiGroup, Version: "v1", Resource: r.resourceName}
+	// Build watch options from the provided resource version
+	watchOptions := metav1.ListOptions{
+		Watch:               true,
+		ResourceVersion:     resourceVersion,
+		AllowWatchBookmarks: true,
+	}
```

Comment:
Move this into `startWatcher` since it's not used anywhere else and doesn't need to be kept around.
```suggestion
```